plugins/node/opentelemetry-instrumentation-aws-sdk/src/services/sqs.ts

```diff
@@ -67,9 +67,12 @@ export class SqsServiceExtension implements ServiceExtension {
           spanAttributes[SemanticAttributes.MESSAGING_OPERATION] =
             MessagingOperationValues.RECEIVE;
 
-          request.commandInput.MessageAttributeNames = (
-            request.commandInput.MessageAttributeNames ?? []
-          ).concat(propagation.fields());
+          request.commandInput.MessageAttributeNames = Array.from(
+            new Set([
+              ...(request.commandInput.MessageAttributeNames ?? []),
+              ...propagation.fields(),
+            ])
+          );
```

Comment:
An optional improvement if you are up to it: since `request.commandInput.MessageAttributeNames` are expected to be unset most of the time, we can move this check to be tested outside of the `Array.from`, and just return  `propagation.fields()` in this case. For these common cases, we won't need to create the interminate arrays and `Set`.